Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs

```diff
@@ -0,0 +1,36 @@
+﻿using AutoMapper;
+using FluentResults;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Streetcode.BLL.DTO.Streetcode;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.DTO.Streetcode.Types;
+using Streetcode.BLL.MediatR.Streetcode.Streetcode.Queries;
+using Streetcode.DAL.Entities.Streetcode.Types;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Streetcode.DAL.Repositories.Interfaces.Streetcode.TextContent;
+
+namespace Streetcode.BLL.MediatR.Streetcode.Streetcode.Handlers.QueryHandlers;
+
+public class GetStreetcodeByIdHandler : IRequestHandler<GetStreetcodeByIdQuery, Result<StreetcodeDTO>>
+{
+    private readonly IMapper _mapper;
+    private readonly IRepositoryWrapper _repositoryWrapper;
+
+    public GetStreetcodeByIdHandler(IRepositoryWrapper repositoryWrapper, IMapper mapper)
+    {
+        _repositoryWrapper = repositoryWrapper;
+        _mapper = mapper;
+    }
+
+    public async Task<Result<StreetcodeDTO>> Handle(GetStreetcodeByIdQuery request, CancellationToken cancellationToken)
+    {
+        var streetcode = await _repositoryWrapper.StreetcodeRepository.GetSingleOrDefaultAsync(
+            predicate: st => st.Id == request.id,
+            include: source => source
+                    .Include(l => l.Tags));
```

Comment:
Please handle the following unhappy paths:
- **SingleOrDefault** may throw an _**InvalidOperationException**_ when there is more than one element that matches predicate; On the other hand, if that condition will never be reached, why we need to iterate over an entire Set?
- If **default value** is returned (__null__), IMapper will throw an exception which will find its way up to the UI. Do we want that behaviour?

Reply:
We have never get more than 1 streetcodes with following id, so why we have to handle this exception?
Added null exception handler

Reply:
Please carefully review the first bullet on the list.

Reply:
done